applications/FemToDemApplication/python_scripts/FemDemMechanicalSolver.py

```diff
@@ -348,16 +337,29 @@ def _execute_after_reading(self):
 
     def _import_constitutive_laws(self):
 
-        if os.path.isfile("materials.py"):
-            # Constitutive law import
-            import KratosMultiphysics.SolidMechanicsApplication.constitutive_law_python_utility as constitutive_law_utils
-            constitutive_law = constitutive_law_utils.ConstitutiveLawUtility(self.main_model_part,
-                                                                             self.main_model_part.ProcessInfo[KratosMultiphysics.SPACE_DIMENSION]);
-            constitutive_law.Initialize();
+        if self.settings["model_import_settings"].Has("path_to_mdpa"):
+            path_to_mdpa = self.settings["model_import_settings"]["path_to_mdpa"].GetString()
+        else:
+            path_to_mdpa = ""
+            
+
+        if path_to_mdpa == "":
+            if os.path.isfile("materials.py"):
+                # Constitutive law import
+                import KratosMultiphysics.SolidMechanicsApplication.constitutive_law_python_utility as constitutive_law_utils
+                constitutive_law = constitutive_law_utils.ConstitutiveLawUtility(self.main_model_part,
+                                                                                self.main_model_part.ProcessInfo[KratosMultiphysics.SPACE_DIMENSION]);
+                constitutive_law.Initialize();
 
+                return True
+            else:
+                return False
+        else:  # test cases
+            materials_path = path_to_mdpa
+            folders = materials_path.split("/")

```

Comment:
Are you sure that the path will always contain "/" and not "\"?

Reply:
As a Windows user I thought about that but in my opinion i would force to use always the "/"

Reply:
I recently started using the [pathlib](https://docs.python.org/3/library/pathlib.html) library (python 3.4) which helps with those issues